avr-hal-generic/src/i2c.rs

```diff
@@ -499,5 +499,82 @@ macro_rules! impl_twi_i2c {
                 Ok(())
             }
         }
+
+        $(#[$i2c_attr])*
```

Comment:
Need clarification on what `$(#[$i2c_attr])*` does. Looks like an attribute that expands?

Reply:
As explained in [my comment](https://github.com/Rahix/avr-hal/issues/90#issuecomment-716125402) this is needed for the doc-string and other type attributes.  This is also recommended by the rust API guidelines as [`C-MACRO-ATTR`](https://rust-lang.github.io/api-guidelines/macros.html#item-macros-compose-well-with-attributes-c-macro-attr).